content/questions/object-clone-stringify-mutation/index.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Object Cloning (JSON.parse + JSON.stringify)
+tags:
+  - object
+  - cloning
+  - mutation
+order: 27
+date: '2019-10-05'
+answers:
+  - 'Joe Nested' // correct
+  - 'Bob Changed'
+  - 'Joe Changed'
+  - 'Bob Nested'
+
+---
+
+Consider objects `a` and `b` below. What gets logged?
+
+```javascript
+const a = { 
+  stringField: 'Joe',
+  nestedField: {field: 'Nested'}
+ };
+const b = JSON.parse(JSON.stringify(a));
+b.stringField = 'Bob'
+b.nestedField.field = 'Changed'
+console.log(a.stringField,
+a.nestedField.field
+);
+```
+
+<!-- explanation -->
+
+Setting `b = JSON.parse(JSON.stringify(a))` will perform a deep copy of object `a`. `b` will not contain any references to `a` and thus mutating any fields in `b` will not affect any fields in `a`.
```

Comment:
Can you add a warning/caveat that this only works for fields that are valid JSON values? I see that you get into tit a bit in the other question you proposed, but I want to make sure it's clear in isolation here. I think MDN has a good discussion about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#Description), maybe consider adding that link in the expanded explanation?